engine/engine.go

```diff
@@ -105,6 +105,8 @@ func NewFromSettings(settings *Settings, flagSet map[string]bool) (*Engine, erro
 		return nil, fmt.Errorf("failed to create script manager. Err: %s", err)
 	}
 
+	b.ExchangeManager = SetupExchangeManager()
+
```

Comment:
This is already set in `Start()`. As it seems like you're using a different workflow to setup/load unsupported exchanges, the `ExchangeManager` is already exported and you should set it up from that context.

Reply:
That's the tricky bit, on my service (using this patch) i go:
```
engine.Bot, err = engine.NewFromSettings(&settings, flagSet)
...
engine.Bot.ExchangeManager.Builder = ExchangeBuilder{}
...
if err = engine.Bot.Start(); err != nil {
```
Main idea here is to have a functional `ExchangeManager` that can be tweaked around before `Start`.
Currently, (pre-patch)  `Start` creates and overwrites any existing `ExchangeManager` and copying the entire contents of `NewFromSettings` and `Start` seems a bit wasteful.

Reply:
When reviewing yesterday it looked like you had left the original `bot.ExchangeManager = SetupExchangeManager()` in `Start()` and I didn't want the two of them there. So long as there is only one, I'm fine with it 👍